database/postgresconn/postgresconn.go

```diff
@@ -182,26 +205,22 @@ func (p *Postgres) Unlock() error {
 }
 
 func (p *Postgres) Run(migration io.Reader) error {
-	if p.config.MultiStatementEnabled {
-		var currentSchema string
-		if err := p.conn.QueryRowContext(context.Background(), "select current_schema()").Scan(&currentSchema); err != nil {
-			return err
-		}
-		fmt.Printf("current schema %v\n", currentSchema)
-		
-		err := multistmt.Parse(migration, multiStmtDelimiter, p.config.MultiStatementMaxSize, p.config.SchemaName, func(m []byte) error {
-			if err := p.runStatement(m); err != nil {
-				return errors.Wrap(err, string(m))
-			}
-			return nil
-		})
-		return err
-	}
-	migr, err := ioutil.ReadAll(migration)
-	if err != nil {
+	var currentSchema string
+	if err := p.conn.QueryRowContext(context.Background(), "select current_schema()").Scan(&currentSchema); err != nil {
 		return err
 	}
-	return p.runStatement(migr)
```

Comment:
Multi-statement, streaming parse is the only supported mode of reading migrations now. This ensures tests always use the same code path to read migrations also, previously they were using different modes in some tests.

Reply:
@evadnoob, thanks for this undocumented change for which we wasted several hours by debugging our strangely failing migration scripts. You might want to update https://github.com/getoutreach/migrate/blob/master/database/postgresconn/README.md?plain=1#L10 as well.

```
create or replace function my_trigger() returns trigger as $body$
...
$body$ language plpgsql;
```

Is a valid syntax too, unfortunately not recognized by the custom parser. We changed it to supported `$$`.